test/Microsoft.AspNet.Mvc.FunctionalTests/Logging/LoggingTests.cs

```diff
@@ -0,0 +1,199 @@
+            Assert.Empty(controller.ActionConstraints);
+            Assert.Empty(controller.RouteConstraints);
+            Assert.Empty(controller.AttributeRoutes);
+        }
+
+        [Fact]
+        public async Task ActionDescriptorValues_LoggedAtStartup()
+        {
+            // Arrange and Act
+            var logs = await GetLogsForStartupAsync();
+            logs = logs.Where(c => c.StateType.Equals(typeof(ActionDescriptorValues)));
+
+            // Assert
+            Assert.Single(logs);
+            dynamic action = logs.First().State;
+            Assert.Equal("Index", action.Name.ToString());
+            Assert.Empty(action.Parameters);
+            Assert.Empty(action.FilterDescriptors);
+            Assert.Equal("action", action.RouteConstraints[0].RouteKey.ToString());
+            Assert.Equal("controller", action.RouteConstraints[1].RouteKey.ToString());
+            Assert.Empty(action.RouteValueDefaults);
+            Assert.Empty(action.ActionConstraints.ToString());
+            Assert.Empty(action.HttpMethods.ToString());
+            Assert.Empty(action.Properties);
+            Assert.Equal("Home", action.ControllerName.ToString());
+        }
+
+        [Fact]
+        public async Task Successful_ActionSelection_Logged()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+            var requestTraceId = Guid.NewGuid().ToString();
+
+            // Act and Assert
+            var response = await client.GetAsync(string.Format("http://localhost/home/index?{0}={1}", 
+                                                        LoggingExtensions.RequestTraceIdQueryKey, requestTraceId));
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var responseData = await response.Content.ReadAsStringAsync();
+            Assert.Equal("Home.Index", responseData);
+
+            //get logs
+            responseData = await client.GetStringAsync("http://localhost/logs");
+            var logActivities = JsonConvert.DeserializeObject<List<ActivityContextDto>>(responseData);
+            var scopeNode = logActivities.FilterByRequestTraceId(requestTraceId)
+                                         .FindScopeByName(nameof(MvcRouteHandler) + ".RouteAsync");
+
+            Assert.NotNull(scopeNode);
+            var logInfo = scopeNode.Messages.Where(logEnty =>
+                                                    logEnty.StateType.Equals(typeof(MvcRouteHandlerRouteAsyncValues)))
+                                            .FirstOrDefault();
+            Assert.NotNull(logInfo);
+            Assert.NotNull(logInfo.State);
+
+            dynamic actionSelection = logInfo.State;
+            Assert.Equal("True", actionSelection.ActionSelected.ToString());
+            Assert.Equal("True", actionSelection.ActionInvoked.ToString());
+            Assert.Equal("True", actionSelection.Handled.ToString());
```

Comment:
Good point...changed these like `Assert.True((bool)actionSelection.ActionSelected);` to work with `dynamic`...